### PR TITLE
updated Instructions on shopping cart project

### PR DESCRIPTION
**Pull request by kaleem99**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by sheenarbw — CHANGES_REQUESTED** [review]
All the functions should take in the data array as an argument. Not just the `get customer baskets` function, ALL the functions.

File: content/projects/understanding-loops/_index.md [hunk 1]
```diff
@@ -119,9 +124,12 @@ Hint: You have already defined some functions that would be useful in finding th
...
+e.g. `getCustomersWithOpenBaskets(dataArray);`
+
 ## Notes to reviewers
 
 - the code must be DRY
 - there can be some global constants to help prevent typos. eg `const DELIVERED="DELIVERED"`
-- functions are supposed to return very specific things
\ No newline at end of file
+- functions are supposed to return very specific things
+- All of your functions stated above should take in at least one argument, the data array, and should work with similar data.
```
Comment (FaithMo) [inline content/projects/understanding-loops/_index.md]: i think maybe rephrase it to "All  the functions.." because these are notes to the reviewer, not the person who wrote the code

**Review by FaithMo — CHANGES_REQUESTED** [review]

**Review by FaithMo — APPROVED** [review]

**Review by Vuyisanani — APPROVED** [review]

File: content/projects/understanding-loops/_index.md [hunk 2]
```diff
@@ -15,30 +15,30 @@ title: Shopping cart calculations
...
 
 Imagine you are working for a hot new online shopping startup. You've seen online shopping, right? Go poke around on Takealot or Amazon if you haven't.
 
-On the online shopping site, users (people who want to buy stuff) navigate around and then add items to their shopping baskets. Once they are happy with their baskets then they "checkout", pay for their goodies, and await their delivery.
+On the online shopping site, users (people who want to buy stuff) navigate around and then add items to their shopping baskets. Once they are happy with their baskets, then they "checkout", pay for their goodies, and await their delivery.
 
 ## Why data matters
 
-As any kind of techie, data will be a part of your life. You're going to need to learn how to make this stuff dance.
+As with any kind of techie, data will be a part of your life. You're going to need to learn how to make this stuff dance.
```
Comment (sheenarbw) [inline content/projects/understanding-loops/_index.md]: techie is short for technical person (sortof). It's a noun. I am a techie. As a techie, I work with data.

Don't just hit things with the Grammarly hammer, make sure that sentences make sense.

File: content/projects/understanding-loops/_index.md [hunk 3]
```diff
@@ -67,20 +67,23 @@ Please note that one person can have multiple baskets. If you look at the data,
 
 ### get baskets belonging to a single customer
 
-Write a function called `get customer baskets` that has an email address as an argument and returns a list/array of all the shopping baskets that belong to the customer with that email address.
+Write a function called `get customer baskets` that takes in an email address and the data array as an argument and returns a list/array of all the shopping baskets that belong to the customer with that email address.
```
Comment (sheenarbw) [inline content/projects/understanding-loops/_index.md]: takes in the email address and the data array as arguments. There are 2 separate arguments

File: content/projects/understanding-loops/_index.md [hunk 4]
```diff
@@ -119,9 +124,12 @@ Hint: You have already defined some functions that would be useful in finding th
...
+e.g. `getCustomersWithOpenBaskets(dataArray);`
+
 ## Notes to reviewers
 
 - the code must be DRY
 - there can be some global constants to help prevent typos. eg `const DELIVERED="DELIVERED"`
-- functions are supposed to return very specific things
\ No newline at end of file
+- functions are supposed to return very specific things
+- All the functions stated above should take in at least one argument, the data array, and should work with similar data.
```
Comment (sheenarbw) [inline content/projects/understanding-loops/_index.md]: This point starts with a capital letter. All the others are small letters. 
Should they all start with capital letters or should they all be small letters?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
some grammer and inconsistency errors

**Review by Sbonelo01 — APPROVED** [review]

**Review by FaithMo — APPROVED** [review]

**Review by ry-oc — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Comment by sheenarbw** [comment]
Merge conflicts

File: content/projects/understanding-loops/_index.md [hunk 5]
```diff
@@ -67,21 +67,24 @@ Please note that one person can have multiple baskets. If you look at the data,
...
 
-Write a function called `required stock` that gets all the items that need to be sent out to delivery.
-
+Write a function called `required stock`, your function should take the data array as an argument and should return all the items that need to be sent out for delivery.
 You need to return data in the correct format. Just include the names and quantities of the items.
 
-For example: if one customer paid for 2 hamsters and another customer paid for one hamster and a bag of sawdust then your function should return the following data structure:
+e.g. `requiredStock(dataStore);`
+
+For example, if one customer paid for 2 hamsters and another customer paid for one hamster and a bag of sawdust then your function should return the following data structure:
```
Comment (ry-oc) [inline content/projects/understanding-loops/_index.md:87]: There may be a missing colon on line 92, please check it out?

**Review by ry-oc — CHANGES_REQUESTED** [review]
one tiny thing missed

**Review by ry-oc — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]
